src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/AbstractFileWriter.cs

```diff
@@ -372,6 +372,38 @@ protected static bool IsKeyword(string name)
             }
         }
 
+        protected List<Kind> GetKindsOfFieldOrNearestParent(TreeType nd, Field field)
+        {
+            if (!IsOverride(field) || field.Kinds.Count > 0)
+            {
+                return field.Kinds;
+            }
+
+            while (IsOverride(field))
+            {
+                nd = Tree.Types.Single(t => t.Name == nd.Base);
```

Comment:
It looks like `_typeMap` may be provided for the purpose of looking up types by name

Reply:
Actually `AbstractFileWriter.GetTreeType()` or `SourceWriter.TryGetBaseType()` (if pulled up to the base type) both seem suitable for this lookup. Perhaps `GetTreeType(nd.Base)` is more straightforward.

Reply:
Thanks. It looks a lot better :heart: